### PR TITLE
add api module

### LINKED ISSUE #36: add API submodule
API submodule should have:

- upload datasets to dataverse
- download datasets from dataverse and dataset generation
- command line communication with the user

### PR DESCRIPTION
**Pull request by atrisovic**
closes #36

### CODE REVIEW

File: spacebench/api/README.md
Comment (mauriciogtec) [inline spacebench/api/README.md:1]: Why do we need the README on the API?

File: spacebench/api/cli.py
Comment (mauriciogtec) [inline spacebench/api/cli.py:1]: This code needs to be contained inside `if __name__ == "__main__"` to be it import safe.
Also, if we support a CLI, it should be included as an entry point in `setup.py`. [Here is an example](https://github.com/google-research/arxiv-latex-cleaner/blob/main/setup.py) and we should choose a name (e.g, `spacecli`).

**Review by mauriciogtec — CHANGES_REQUESTED** [review]

**Comment by mauriciogtec** [comment]
I added some comments for review.
Summary of comments:
1) make the CLI import safe
2) add an entrypoint for the CLI (in setup)
3) make a "within Python" API.

File: spacebench/api/README.md
Comment (atrisovic) [reply spacebench/api/README.md:1]: 😅

Comment (atrisovic) [reply spacebench/api/README.md:1]: I saw it here: https://github.com/snap-stanford/ogb/tree/master/ogb/io

File: spacebench/api/cli.py
Comment (atrisovic) [reply spacebench/api/cli.py:1]: I think we should support both cli + normal package functionality. The client is really handy for testing

Comment (atrisovic) [reply spacebench/api/cli.py:1]: Incorporated in #42

**Comment by atrisovic** [comment]
Closing as superseded by #42

**Pull request closed without merging**